draft-kazuho-httpbis-priority.md

```diff
@@ -137,94 +137,21 @@ multiple experiments from independent research have shown that simpler schemes
-connection error of type PROTOCOL_ERROR for HTTP/2 {{!RFC7540}}, or of type
-H3_SETTINGS_ERROR for HTTP/3 {{!I-D.ietf-quic-http}}.
-
-In HTTP/2, the setting SHOULD appear in the first SETTINGS frame and peers
-MUST NOT process the setting if it is received multiple times in order to
-avoid changing the agreed upon prioritization scheme.
-
-If there is a prioritization scheme supported by both the client and server,
-then the server's preference order prevails and both peers SHOULD
-only use the agreed upon priority scheme for the remainder of the session.
-The server chooses because it is in the best position to know what
-information from the client is of the most value.
-
-Once the negotiation is complete, endpoints MAY stop sending hop-by-hop
-prioritization signals that were not negotiated in order to conserve bandwidth.
-However, endpoints SHOULD continue sending end-to-end signals (e.g., the
-Priority header field), as that might have meaningful effect to other nodes that
-handle the HTTP message.
-
-## Defined Prioritization Scheme Values
-
-This document defines two prioritization scheme values for use with the
-SETTINGS_PRIORITIES setting.
-
-### H2_TREE {#settings-h2-scheme}
-
-This document defines the priority scheme identifier H2_TREE (8-bit value of 1)
-that indicates support for HTTP/2-style priorities ({{!RFC7540}}, Section 5.3).
-
-The H2_TREE priority scheme identifier MUST NOT be be sent in an HTTP/3 settings
-because there is no defined mapping of this scheme. Endpoints MUST treat receipt
-of H2_TREE as a connection error of type H3_SETTINGS_ERROR.
-
-### URGENCY {#settings-this-scheme}
-
-This document defines the priority scheme identifier URGENCY (8-bit value of 2)
-that indicates support for the extensible priority scheme defined in the present
-document.
-
-An intermediary connecting to a backend server SHOULD declare support for the
-extensible priority scheme when and only when all the requests that are to be
-sent on that backend connection originates from one client-side connection that
-has negotiated the use of the extensible priority scheme (see {{fairness}}).
+## Disabling HTTP/2 Priorities
+The problems and insights set out above are motivation for allowing endpoints to
+opt out of using the HTTP/2 priority scheme. Endpoints would benefit from
+understanding their peer's intention, so the new
+SETTINGS_DISABLE_HTTP2_PRIORITIES is defined. The value of the parameter MUST be
+0 or 1.
+
+An HTTP/2 client SHOULD use the HTTP/2 priority scheme (as it sees fit) until it
+receives the peer's SETTINGS_DISABLE_HTTP2_PRIORITIES settings with the value of
+`1`. As the SETTINGS frame precedes any priority signal sent from a client, a server can
+determine if it should respect the HTTP/2 scheme before building state.
+
+A client that disables the HTTP/2 priority scheme SHOULD subtitute it with some
+other means of signaling request priority. The scheme extensible priority scheme
+is RECOMMENDED.
```

Comment:
The client SHOULD use the extensible priority scheme until it finds out that the server has deprecated the HTTP/2 priority scheme. It MAY continue sending the Priority header field even after it learns that the server is using the HTTP/2 priority frame, as header fields are end-to-end signal.

Reply:
Done in 40e9875.